transports/daily/src/transport.ts

```diff
@@ -24,34 +24,107 @@ import {
   logger,
 } from "@pipecat-ai/client-js";
 
+import {
+  WavMediaManager,
+} from "@pipecat-ai/realtime-websocket-transport";
+
 export interface DailyTransportAuthBundle {
   room_url: string;
   token: string;
 }
 
 export interface DailyTransportConstructorOptions {
+  enableEarlyMicAccess?: boolean;
```

Comment:
i feel like the name `earlyMicAccess` is misleading. when i read this initially, i thought "why can't we just start the mic before joining the call?" -- but it's not about starting the mic early, it's about capturing the audio from the mic and buffering/recording it so that those bytes aren't lost while waiting for the bot to join.  Maybe `bufferLocalAudio` or a more verbose `bufferLocalAudioUntilReady` with events `audio-buffering-started/stopped`?

Reply:
I like the idea of using `enable` to convey that we are activating something that is not enabled by default. And the reason to use `early`, was to give the sense that it is happening before we are fully ready.

But yeah, I agree that `enableEarlyMicAccess` was probably not descriptive enough.   =(

But when I think about buffer, it feels like the SDK is going to buffer, but I still need to do something with that later. 

What do you think about `enableEarlyMicCapture` instead ? In this case the events would be `EARLY_MIC_CAPTURE_STARTED` and `EARLY_MIC_CAPTURE_STOPPED`.

or `enableSpeakBeforeReady`, In this case the events would be `SPEAK_BEFORE_READY_STARTED` and `SPEAK_BEFORE_READY_STOPPED`.

Reply:
Decided during chat:
`bufferLocalAudioUntilBotReady` with events `audio-buffering-started/stopped`

Reply:
Done!